acceptance/launcher_test.go

```diff
@@ -78,10 +77,54 @@ func testLauncher(t *testing.T, when spec.G, it spec.S) {
 			expected := "worker-process-val"
 			assertOutput(t, cmd, expected)
 		})
+	})
+
+	when("process is direct=false", func() {
+		when("the process type has no args", func() {
+			it("runs command as script", func() {
+				h.SkipIf(t, runtime.GOOS == "windows", "scripts are unsupported on windows")
+				cmd := exec.Command("docker", "run", "--rm",
+					"--env", "VAR1=val1",
+					"--env", "VAR2=val with space",
+					launchImage, "indirect-process-with-script",
+				)
+				assertOutput(t, cmd, "'val1' 'val with space'")
+			})
+		})
+
+		when("the process type has args", func() {
+			when("buildpack API 0.4", func() {
+				it("command and args become shell-parsed tokens in a script", func() {
+					var val2 string
+					if runtime.GOOS == "windows" {
+						val2 = `"val with space"` //windows values with spaces must contain quotes
+					} else {
+						val2 = "val with space"
+					}
+					cmd := exec.Command("docker", "run", "--rm",
+						"--env", "VAR1=val1",
+						"--env", "VAR2="+val2,
+						launchImage, "indirect-process-with-args",
+					)
+					assertOutput(t, cmd, "'val1' 'val with space'")
+				})
+			})
+
+			when("buildpack API < 0.4", func() {
+				it("args become arguments to bash", func() {
+					h.SkipIf(t, runtime.GOOS == "windows", "scripts are unsupported on windows")
+					cmd := exec.Command("docker", "run", "--rm",
+						launchImage, "legacy-indirect-process-with-args",
+					)
+					assertOutput(t, cmd, "'arg' 'arg with spaces'")
+				})
+			})
+		})
 
 		it("sources scripts from process specific directories", func() {
-			cmd := exec.Command("docker", "run", "--rm", launchImage, "worker")
-			expected := "sourced bp profile\nsourced bp worker profile\nsourced app profile"
+			h.SkipIf(t, runtime.GOOS == "windows", "profile scripts are broken on windows?")
```

Comment:
@micahyoung @ameyer-pivotal

I added this test to ensure variables set in the profile scripts are available when the command runs. It doesn't pass on windows. I believe this is bug with windows profiles scripts but I would appreciate your input.

Reply:
Very possible there's a bug, since it's so new. It seemed to be working when we implemented it, but I believe only unit tests existed at the time